cmd/admin-policy-unset.go

```
@@ -18,120 +18,22 @@
+	OnUsageError:       onUsageError,
+	Before:             setGlobalsFromContext,
+	HideHelpCommand:    true,
+	Hidden:             true,
+	CustomHelpTemplate: `Please use 'mc admin policy detach'`,
 }
 
-// mainAdminPolicyUnset is the handler for "mc admin policy unset" command.
-func mainAdminPolicyUnset(ctx *cli.Context) error {
-	checkAdminPolicyUnsetSyntax(ctx)
-
-	console.SetColor("PolicyMessage", color.New(color.FgGreen))
-	console.SetColor("Policy", color.New(color.FgBlue))
-
-	// Get the alias parameter from cli
-	args := ctx.Args()
-	aliasedURL := args.Get(0)
-	policiesToUnset := args.Get(1)
-	entityArg := args.Get(2)
-
-	userOrGroup, isGroup, e1 := parseEntityArg(entityArg)
-	fatalIf(probe.NewError(e1).Trace(args...), "Bad last argument")
-
-	// Create a new MinIO Admin Client
-	client, err := newAdminClient(aliasedURL)
-	fatalIf(err, "Unable to initialize admin connection.")
-
-	var existingPolicies string
-
-	if !isGroup {
-		userInfo, e := client.GetUserInfo(globalContext, userOrGroup)
-		fatalIf(probe.NewError(e).Trace(args...), "Unable to get user policy info")
-		existingPolicies = userInfo.PolicyName
-	} else {
-		groupInfo, e := client.GetGroupDescription(globalContext, userOrGroup)
-		fatalIf(probe.NewError(e).Trace(args...), "Unable to get group policy info")
-		existingPolicies = groupInfo.Policy
-	}
-
-	newPolicies, e := removeCannedPolicies(existingPolicies, policiesToUnset)
-	if e != nil {
-		fatalIf(probe.NewError(e).Trace(args...), "Unable to unset the policy")
-	}
-
-	e = client.SetPolicy(globalContext, newPolicies, userOrGroup, isGroup)
-	if e == nil {
-		printMsg(userPolicyMessage{
-			op:          ctx.Command.Name,
-			Policy:      policiesToUnset,
-			UserOrGroup: userOrGroup,
-			IsGroup:     isGroup,
-		})
-	} else {
-		fatalIf(probe.NewError(e), "Unable to unset the policy")
-	}
+func mainAdminPolicyUnsetErr(ctx *cli.Context) error {
+	console.Infoln("Please use 'mc admin policy detach'")
```

This function should return an error, and ensure that mc exits with non-zero exit status.